src/buffer_manager.ts

```diff
@@ -413,40 +428,46 @@ export class BufferManager implements Disposable {
     // #region Sync layout
 
     private onEditorLayoutChanged = async () => {
-        if (!this.syncEditorLayoutPromise) {
-            this.syncEditorLayoutPromise = new ManualPromise();
+        this.syncLayoutPromise = this.syncLayoutPromise ?? new ManualPromise();
+        this.isLayoutOutdated = true;
+        this.syncLayoutSource?.cancel();
+        this.syncLayoutSource = new CancellationTokenSource();
+        if (!this.isSyncingLayout) {
+            await this.syncEditorLayoutDebounced();
```

Comment:
When we skip this because `isSyncingLayout=true`, is it possible that we lose the "latest" update? Or does the while-loop in `syncEditorLayout()` always get the latest?

Reply:
When an event occurs, `isLayoutOutdated` will be set to true. The ongoing loop will start a new cycle and use the latest data.